theano/sandbox/cuda/blas.py

```diff
@@ -216,7 +216,8 @@ def infer_shape(self, node, shapes):
         xshp, yshp = shapes
         return [xshp[:-1] + yshp[2:]]
 
-batched_dot = BatchedDotOp()
+batched_dot = GpuBatchedDot()
+BatchedDotOp = GpuBatchedDot()
 """
 Call cublasSgemmBatched. Take 2 3d tensor as input.
 """
```

Comment:
I added a fix in https://github.com/Theano/Theano/pull/4045

thanks

On Wed, Feb 10, 2016 at 11:37 AM, Tim Cooijmans <notifications@github.com>
wrote:

> In theano/sandbox/cuda/blas.py
> <https://github.com/Theano/Theano/pull/3947#discussion_r52484129>:
>
> > @@ -216,7 +216,8 @@ def infer_shape(self, node, shapes):
> >          xshp, yshp = shapes
> >          return [xshp[:-1] + yshp[2:]]
> >
> > -batched_dot = BatchedDotOp()
> > +batched_dot = GpuBatchedDot()
> > +BatchedDotOp = GpuBatchedDot()
> >  """
> >  Call cublasSgemmBatched. Take 2 3d tensor as input.
> >  """
>
> I think two things went wrong here -- batched_dot is now separated from
> its docstring and GpuBatchedDot shouldn't be instantiated before
> assigning it to BatchedDotOp. @nouiz <https://github.com/nouiz>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3947/files#r52484129>.
>